source/extensions/clusters/aggregate/cluster_lb.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/upstream/load_balancer_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+class AggregateLoadBalancerContext : public Upstream::LoadBalancerContext {
+public:
+  AggregateLoadBalancerContext(Upstream::LoadBalancerContext* context,
+                               Upstream::LoadBalancerBase::HostAvailability host_availability,
+                               uint32_t host_priority)
+      : host_availability_(host_availability), host_priority_(host_priority) {
+    if (context == nullptr) {
+      owned_context_ = std::make_unique<Upstream::LoadBalancerContextBase>();
+      context_ = owned_context_.get();
+    } else {
+      context_ = context;
+    }
+  }
+
+  // Upstream::LoadBalancerContext
+  absl::optional<uint64_t> computeHashKey() override { return context_->computeHashKey(); }
+  const Network::Connection* downstreamConnection() const override {
+    return context_->downstreamConnection();
+  }
+  const Router::MetadataMatchCriteria* metadataMatchCriteria() override {
+    return context_->metadataMatchCriteria();
+  }
+  const Http::HeaderMap* downstreamHeaders() const override {
+    return context_->downstreamHeaders();
+  }
+  const Upstream::HealthyAndDegradedLoad&
+  determinePriorityLoad(const Upstream::PrioritySet&,
+                        const Upstream::HealthyAndDegradedLoad& original_priority_load) override {
+    // Re-assign load. Set all traffic to the priority and availability selected in aggregate
+    // cluster.
+    // TODO(yxue): allow determinePriorityLoad to affect the load of top level cluster and verify it
```

Comment:
probably want to note that this won't work right in the docs. I think the only limitation is with `PriorityLoad` retry plugins